### PR TITLE
Respect the `Le` field of commands

### PR DESCRIPTION
**Pull request by sosthene-nitrokey**
This patch fixes handling of command apdus with `Le < 255`. The current behavior was to use response chaining with `255` as the limit for any response. This PR changes this to the correct behavior to use response chaining when the response is larger than `Le` particularly when `Le` is lower than 255. 

Relevant parts from iso 7816-4: 

> The Le field encodes Ne.

> Nr denotes the  number of bytes in the response data field. Nr shall be  less than or  equal to Ne. 

This PR depends on [iso7816#](https://github.com/ycrypto/iso7816/pull/4) otherwise the tests fail.  Once it is merged and a new version of the `iso7816` crate is released we will be able to remove the git dependency in `Cargo.toml`. Currently the tests leave the `Le` field absent, indicating `Ne = 0`, which was ignored. This PR makes the test commands contain `Le = 0x00` indicating `Ne = 256`. It also uses [`hex-literal`](https://lib.rs/crates/hex-literal) to make the test data easier to read.

### CODE REVIEW

File: src/dispatch.rs [hunk 1]
```diff
@@ -290,8 +294,8 @@ impl ApduDispatch
 
                 if self.was_request_chained || res.len() > interchanges::SIZE {
 
-                    // Send 256 bytes max at a time.
-                    let boundary = core::cmp::min(256, res.len());
+                    // Do not send more than the expcted bytes
```
Comment (robin-nitrokey) [inline src/dispatch.rs:297]: typo

**Review by robin-nitrokey — APPROVED** [review]
Thanks for the fix, looks good to me!

File: src/dispatch.rs [hunk 2]
```diff
@@ -290,8 +294,8 @@ impl ApduDispatch
 
                 if self.was_request_chained || res.len() > interchanges::SIZE {
 
-                    // Send 256 bytes max at a time.
-                    let boundary = core::cmp::min(256, res.len());
+                    // Do not send more than the expected bytes
+                    let boundary = core::cmp::min(self.response_len_expected, res.len());
```
Comment (robin-nitrokey) [inline src/dispatch.rs:298]: Hmm, one afterthought:  Shouldn’t we take into account `interchanges::SIZE` here?  `self.response_len_expected` could be up to `u16::MAX + 1`, `res.len()` up to `crate::response::SIZE`.  `interchanges::SIZE` is smaller than that.

File: src/dispatch.rs (hunk 1 above)
Comment (robin-nitrokey) [reply src/dispatch.rs]: Also, please update the comment in line 284.  (I can’t comment on it directly as it’s not part of the diff.)

File: src/dispatch.rs [hunk 3]
```diff
@@ -288,10 +300,10 @@ impl ApduDispatch
             }
             RawApduBuffer::Response(res) => {
 
-                if self.was_request_chained || res.len() > interchanges::SIZE {
+                if self.was_request_chained || res.len() > MAX_INTERCHANGE_DATA {
 
-                    // Send 256 bytes max at a time.
-                    let boundary = core::cmp::min(256, res.len());
+                    // Do not send more than the expected bytes
+                    let boundary = self.response_len_expected.min(res.len()).min(MAX_INTERCHANGE_DATA);
```
Comment (nickray) [inline src/dispatch.rs:306]: :eyes: min on numbers

**Comment by nickray** [comment]
I force-pushed to rebase on the merged clippy-warning-fix.

**Pull request merged**